### ISSUE TITLE
Exception thrown when calling forgot password with --production flag and MAIL_URL set

### ISSUE THREAD

**Issue by santiagopuentep** [meteor/meteor#11709]
On Meteor 2.4, calling `Accounts.forgotPassword` with the `--production` flag active and `MAIL_URL` is set with real keys throws this exception:

```
Exception while invoking method 'forgotPassword' TypeError: Cannot read property '_syncSendMail' of null
    at smtpSend (packages/email/email.js:154:13)
    at Object.Email.send (packages/email/email.js:233:5)
    at Object.Email.send (packages/mdg:meteor-apm-agent/lib/hijack/email.js:10:28)
    at AccountsServer.Accounts.sendResetPasswordEmail (packages/accounts-password/password_server.js:667:9)
    at MethodInvocation.forgotPassword (packages/accounts-password/password_server.js:478:12)
    at MethodInvocation.methodMap.<computed> (packages/mdg:meteor-apm-agent/lib/hijack/wrap_session.js:161:30)
    at maybeAuditArgumentChecks (packages/ddp-server/livedata_server.js:1885:12)
    at packages/ddp-server/livedata_server.js:769:19
    at Meteor.EnvironmentVariable.EVp.withValue (packages\meteor.js:1257:12)
    at packages/ddp-server/livedata_server.js:767:46
    at Meteor.EnvironmentVariable.EVp.withValue (packages\meteor.js:1257:12)
    at packages/ddp-server/livedata_server.js:765:46
    at new Promise (<anonymous>)
    at Session.method (packages/ddp-server/livedata_server.js:739:23)
    at packages/mdg:meteor-apm-agent/lib/hijack/wrap_session.js:52:38
    at Meteor.EnvironmentVariable.EVp.withValue (packages\meteor.js:1257:12)
    at Session.sessionProto.protocol_handlers.method (packages/mdg:meteor-apm-agent/lib/hijack/wrap_session.js:51:44)
    at packages/ddp-server/livedata_server.js:603:43
```

`smtpSend()` receives no transport, and there's no check of existence before calling one of it's function.

### Expected behavior:
No exception should be thrown.

**Comment by megawebmaster** [comment]
@StorytellerCZ - I believe this is caused by missing mail setup in production mode. In this commit https://github.com/meteor/meteor/commit/c704a16935f92685b87b9a3b8d6252ee8b981a10 there is a change that removed check for transport being non-null before calling `smptSend`.

Do you know why was this change introduced?

**Comment by santiagopuentep** [comment]
Sorry, I didn't see the notifications for this. How do I test the code?

**Comment by DerGut** [comment]
We are getting the same error from a `Accounts.sendVerificationEmail` call:
```
TypeError: Cannot read property '_syncSendMail' of null
 at smtpSend (packages/email/email.js:154:13)
 at Object.Email.send (packages/email/email.js:233:5)
 at AccountsServer.Accounts.sendVerificationEmail (packages/accounts-password/password_server.js:687:9)
```

Interestingly, this error only occurs in our CI test jobs running `meteor test --once --driver-package meteortesting:mocha` without any environment. We weren't yet able to replicate it locally.

**Comment by megawebmaster** [comment]
@DerGut - the same happens to me and it is because Meteor changed how they check for mail transport. If you run it in a production-like env it is enough, so if you don't supply a correct `MAIL_URL` then you'll get this error.

As I'm using Docker Compose to run the test I just added another service (mailcatcher) and pointed Meteor mailer there. It was enough to make it work :wink:

**Comment by DerGut** [comment]
@megawebmaster Thanks for the info! But I don't even understand why `smtpSend` is invoked instead of `devModeSend` in the first place. Looking at https://github.com/meteor/meteor/blob/081a7792deb23389f3cac8486558267b265f8990/packages/email/email.js#L231-L236, none of `Meteor.isProduction`, `MAIL_URL` or `Meteor.settings.packages?.email` should be truthy in my case 🤔 .

At least I don't run the tests with a `--production` flag or have any `NODE_ENV=production` or `MAIL_URL=<some url>` environment variables set.

**Comment by denihs** [comment]
Hi,

The fix for this is already out in version 2.7-rc.0. :raised_hands: 

I'm closing this issue now.
